realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -190,4 +194,23 @@ public boolean isPartialRealm(RealmConfiguration configuration) {
     public void addSupportForObjectLevelPermissions(RealmConfiguration.Builder builder) {
         builder.addModule(new ObjectPermissionsModule());
     }
+
+    @Override
+    @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
+    public void waitForNetworkThreadExecutorToFinish() {
```

Comment:
I like the idea, however, this is used only in tests, I prefer using this logic inside `SyncTestUtils` (similar to what we did with accessing the refresh token with reflection)

Reply:
Fair enough. I'll move the implementation